libleptongui/scheme/schematic/builtins.scm

```diff
@@ -525,28 +525,93 @@
   (x_dialog_hotkeys (*current-window)))
 
 (define-action-public (&options-grid #:label (G_ "Switch Grid Style"))
-  (run-callback i_callback_options_grid "&options-grid"))
+  (define *window (*current-window))
+  (define *options (schematic_window_get_options *window))
+
+  (gschem_options_cycle_grid_mode *options)
+
+  (let ((grid-mode (pointer->string
+                    (schematic_grid_mode_to_string
+                     (gschem_options_get_grid_mode *options)))))
+    (match grid-mode
+      ("none" (log! 'message (G_ "Grid OFF")))
+      ("dots" (log! 'message (G_ "Dot grid selected")))
+      ("mesh" (log! 'message (G_ "Mesh grid selected")))
+      (_ (error "Invalid grid mode: ~S" grid-mode)))))
+
 
 (define-action-public (&options-snap #:label (G_ "Switch Snap Mode"))
-  (run-callback i_callback_options_snap "&options-snap"))
+  (define *window (*current-window))
+  (define *options (schematic_window_get_options *window))
+
+  (gschem_options_cycle_snap_mode *options)
+
+  (let ((snap-mode (pointer->string
+                    (schematic_snap_mode_to_string
+                     (gschem_options_get_snap_mode *options)))))
+    (match snap-mode
+      ("off" (log! 'message (G_ "Snap OFF (CAUTION!)")))
+      ("grid" (log! 'message (G_ "Snap ON")))
+      ("resnap" (log! 'message (G_ "Snap back to the grid (CAUTION!)")))
```

Comment:
These messages are not printed to the log
if the snap mode is set by clicking
on the "snap size" status bar label.

Reply:
Added a `FIXME` comment on this, thank you!